server/jobs/statusPoller.js

```diff
@@ -12,53 +14,102 @@ if (parentPort) {
-      const WUstate = wuResult.Workunit.State;
+        });
+        const WUstate = wuResult.Workunit.State;
+
+        //update JobExecution
+        if(WUstate === 'completed' || WUstate === 'wait' || WUstate === 'blocked' || WUstate === 'failed') {
+          const newjobExecution = { status: WUstate, wu_duration : wuResult.Workunit.TotalClusterTime || null };
+          const result = await jobExecution.update(newjobExecution,{where:{id:jobExecution.id}});
+          logToConsole(`✔️  JOB EXECUTION GOT UPDATED, ("${jobExecution.job.name}") ${result.wuid} = ${result.status} ${result.status === 'completed' ? "👍" : "🚩🚩🚩"}`);
+          logToConsole(result.toJSON());
+          if(WUstate === 'completed' || WUstate === 'failed'){
+            let workFlowURL, dataflow;
+            const cluster = await Cluster.findOne({ where: { id: jobExecution.clusterId } });
+            wuURL = `${cluster.thor_host}:${cluster.thor_port}/?Wuid=${wuResult.Workunit.Wuid}&Widget=WUDetailsWidget`;
 
-      //update JobExecution
-      if(WUstate === 'completed' || WUstate === 'wait' || WUstate === 'blocked' || WUstate === 'failed') {              
-        const newjobExecution = { status: WUstate, wu_duration : wuResult.Workunit.TotalClusterTime || null };
-        const result = await jobExecution.update(newjobExecution,{where:{id:jobExecution.id}}); 
-        logToConsole(`✔️  JOB EXECUTION GOT UPDATED, ("${jobExecution.job.name}") ${result.wuid} = ${result.status} ${result.status === 'completed' ? "👍" : "🚩🚩🚩"}`);
-        logToConsole(result.toJSON());
-        if(WUstate === 'completed' || WUstate === 'failed'){
-           if(WUstate === 'completed'){
+            if(jobExecution.dataValues?.dataflowId){ // If job part of workflow build a URL
+                  dataflow = await Dataflow.findOne({ where: { id: jobExecution.dataValues?.dataflowId } });
+                  workFlowURL = `${process.env.WEB_URL}${dataflow.application_id}/dataflowinstances/dataflowInstanceDetails/${dataflow.id}/${jobExecution.jobExecutionGroupId}`
+            }
+             
+            if(WUstate === 'completed'){
               logToConsole(`🔍  WORKER_THREAD IS DONE, PASSING CHECKING ON DEPENDING JOBS TO MAIN THREAD, "${jobExecution.job.name}" - ${jobExecution.wuid} - ${jobExecution.jobId}...`);
-              // will trigger JobScheduler.scheduleCheckForJobsWithSingleDependency on main thread.
-              dispatchAction('scheduleDependentJobs',{ dependsOnJobId: jobExecution.jobId, dataflowId: jobExecution.dataflowId, jobExecutionGroupId: jobExecution.jobExecutionGroupId });
-              // await JobScheduler.scheduleCheckForJobsWithSingleDependency({ dependsOnJobId: jobExecution.jobId, dataflowId: jobExecution.dataflowId });    
+              //If the job just completed is  a part of a workflow + notification not set up at workflow level -> send notification if set at job level
+              if(dataflow && dataflow?.metaData?.notification?.notify === 'Never' ){
```

Comment:
Same for this code? Cant we move this condition checks to the workflowUtil?

Reply:
Moved.